sklearn/metrics/tests/test_common.py

```diff
@@ -1140,7 +1152,7 @@ def check_sample_weight_invariance(name, metric, y1, y2):
     weighted_score = metric(y1, y2, sample_weight=sample_weight)
 
     # use context manager to supply custom error message
-    with assert_raises(AssertionError) as cm:
+    with pytest.raises(AssertionError) as cm:
```

Comment:
This is the same issue than above. You need to raise a ValueError within the context manager.